srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,153 @@
+
+        if 'behind_vehicle_distance' in config.other_parameters:
+            self._behind_vehicle_distance = float(
+                config.other_parameters['behind_vehicle_distance']['value'])
+        else:
+            self._behind_vehicle_distance = 5  # m
+
+        self._end_distance = self._front_vehicle_distance + 5
+
+        if 'parking_lane_side' in config.other_parameters:
+            self._parking_lane_side = config.other_parameters['parking_lane_side']['value']
+        else:
+            self._parking_lane_side = "right"
+
+        super(ParkingExit, self).__init__("ParkingExit",
+                                          ego_vehicles,
+                                          config,
+                                          world,
+                                          debug_mode,
+                                          criteria_enable=criteria_enable)
+
+    def _spawn_blocking_vehicle(self, spawn_point):
+        CarlaDataProvider.request_new_actor(
+            'vehicle.tesla.model3', spawn_point)
+
+    def _create_behavior(self):
+        """
+        Deactivate OutsideRouteLanesTest, then move ego to the parking point, generate blocking vehicles in front of and behind the ego.
+        After ego drives away, actviate OutsideRouteLanesTest, end scenario.
+        """
+
+        sequence = py_trees.composites.Sequence()
+
+        # Put blocking vehicles
+        if self._parking_lane_side == "left":
+            parking_waypoint = self._map.get_waypoint(
+                self.ego_vehicles[0].get_location()).get_left_lane()
+        else:
+            parking_waypoint = self._map.get_waypoint(
+                self.ego_vehicles[0].get_location()).get_right_lane()
+
+        front_points = parking_waypoint.next(self._front_vehicle_distance)
+        if len(front_points) > 0:
+            CarlaDataProvider.request_new_actor(
+                'vehicle.tesla.model3', front_points[0].transform)
+
+        behind_points = parking_waypoint.previous(
+            self._behind_vehicle_distance)
+        if len(behind_points) > 0:
+            CarlaDataProvider.request_new_actor(
+                'vehicle.tesla.model3', behind_points[0].transform)
+
+        # Deactivate OutsideRouteLanesTest
+        sequence.add_child(SwitchOutsideRouteLanesTest(False))
+
+        # Teleport ego to the parking point
+        sequence.add_child(ActorTransformSetter(
+            self.ego_vehicles[0], parking_waypoint.transform))
+
+        root = py_trees.composites.Parallel(
+            policy=py_trees.common.ParallelPolicy.SUCCESS_ON_ONE)
+
+        end_condition = py_trees.composites.Sequence()
+        end_condition.add_child(InTriggerDistanceToLocation(
+            self.ego_vehicles[0], parking_waypoint.transform.location, self._end_distance, operator.gt, name="EndTrigger"))
+        root.add_child(end_condition)
+        sequence.add_child(root)
+
+        sequence.add_child(SwitchOutsideRouteLanesTest(True))
+
```

Comment:
Just a comment here, there is no behavior that destroys the actor after the scenario, so they'll stay for the rest of the simulation. If this is intended (it doesn't matter either way), then so be it

Reply:
I think there should be less side effect, so I add ActorDestroy to remove them.